bigquery/google/cloud/bigquery/magics.py

```diff
@@ -494,6 +495,8 @@ def _cell_magic(line, query):
         args.use_bqstorage_api or context.use_bqstorage_api, context.credentials
     )
 
+    close_transports = functools.partial(_close_transports, client, bqstorage_client)
```

Comment:
It's pretty easy to forget to call this if we add any other conditions to this magic.

Let's put everything in a `try` block and make sure we call `close_transports` in the `finally` clause.

Reply:
On second thought, I think that actually makes more sense, will change. :+1: 

I initially juggled with the same idea, but felt that having an enormous try-finally block is somewhat odd. However, it's because the method itself is so long and needs refactoring, and making it harder to accidentally re-introduce the leak is more important.